dev/com.ibm.ws.jpa_eclipselink_fat/fat/src/com/ibm/ws/jpa/TestOLGH8461_EJB.java

```diff
@@ -0,0 +1,192 @@
+        PrivHelper.generateCustomPolicy(server, FATSuite.JAXB_PERMS);
+        bannerStart(TestOLGH8461_EJB.class);
+        timestart = System.currentTimeMillis();
+
+        int appStartTimeout = server.getAppStartTimeout();
+        if (appStartTimeout < (120 * 1000)) {
+            server.setAppStartTimeout(120 * 1000);
+        }
+
+        int configUpdateTimeout = server.getConfigUpdateTimeout();
+        if (configUpdateTimeout < (120 * 1000)) {
+            server.setConfigUpdateTimeout(120 * 1000);
+        }
+
+        server.startServer();
+
+        setupDatabaseApplication(server, RESOURCE_ROOT + "ddl/");
+
+        final Set<String> ddlSet = new HashSet<String>();
+
+        System.out.println("TestOLGH8461_EJB Setting up database tables...");
+
+        ddlSet.clear();
+        for (String ddlName : dropSet) {
+            ddlSet.add(ddlName.replace("${dbvendor}", getDbVendor().name()));
+        }
+        executeDDL(server, ddlSet, true);
+
+        ddlSet.clear();
+        for (String ddlName : createSet) {
+            ddlSet.add(ddlName.replace("${dbvendor}", getDbVendor().name()));
+        }
+        executeDDL(server, ddlSet, false);
+
+        setupTestApplication();
+    }
+
+    private static void setupTestApplication() throws Exception {
+        JavaArchive ejbApp = ShrinkWrap.create(JavaArchive.class, "olgh8461Ejb.jar");
+        ejbApp.addPackages(true, "com.ibm.ws.jpa.olgh8461.ejblocal");
+        ejbApp.addPackages(true, "com.ibm.ws.jpa.olgh8461.model");
+        ejbApp.addPackages(true, "com.ibm.ws.jpa.olgh8461.testlogic");
+        ShrinkHelper.addDirectory(ejbApp, RESOURCE_ROOT + "ejb/olgh8461Ejb.jar");
+
+        WebArchive webApp = ShrinkWrap.create(WebArchive.class, appName + ".war");
+        webApp.addPackages(true, "com.ibm.ws.jpa.olgh8461.ejb");
+        ShrinkHelper.addDirectory(webApp, RESOURCE_ROOT + "ejb/" + appName + ".war");
+
+        final JavaArchive testApiJar = buildTestAPIJar();
+
+        final EnterpriseArchive app = ShrinkWrap.create(EnterpriseArchive.class, appNameEar);
+        app.addAsModule(ejbApp);
+        app.addAsModule(webApp);
+        app.addAsLibrary(testApiJar);
+        ShrinkHelper.addDirectory(app, RESOURCE_ROOT + "ejb", new org.jboss.shrinkwrap.api.Filter<ArchivePath>() {
+
+            @Override
+            public boolean include(ArchivePath arg0) {
+                if (arg0.get().startsWith("/META-INF/")) {
+                    return true;
+                }
+                return false;
+            }
+
+        });
+
+        ShrinkHelper.exportToServer(server, "apps", app);
+
+        Application appRecord = new Application();
+        appRecord.setLocation(appNameEar);
+        appRecord.setName(appName);
+        ConfigElementList<ClassloaderElement> cel = appRecord.getClassloaders();
+        ClassloaderElement loader = new ClassloaderElement();
+        loader.setApiTypeVisibility("+third-party");
+//        loader.getCommonLibraryRefs().add("HibernateLib");
```

Comment:
Probably should delete the Hibernate line, even though it is just a comment, for the eclipselink bucket, to avoid confusion for anyone reading through it.